engine/src/main/java/com/datatorrent/stram/StramClient.java

```diff
@@ -455,6 +462,13 @@ public void startApplication() throws YarnException, IOException
       }
       String libJarsCsv = copyFromLocal(fs, appPath, localJarFiles.toArray(new String[]{}));
 
+      // Now that the extra libs are copied, delete if it is required
+      for (Map.Entry<String, Boolean> jars : extraResources.entrySet()) {
+        if (jars.getValue()) {
+          FileUtils.deleteQuietly(new File(jars.getKey()));
```

Comment:
Why quietly? If a user added a temporary jar and specified that it needs to be deleted, should not warning be at minimum logged if the file can not be deleted for any reason?